### PR TITLE
Restarting: option to keep wsv data

### PR DESCRIPTION
**Pull request by MBoldyrev**
### Description of the Change

This change adds WSV schema versioning and an option to reuse existing WSV state at startup.

### CODE REVIEW

File: irohad/main/impl/pg_connection_init.cpp [hunk 1]
```diff
@@ -5,16 +5,164 @@
...
+   *    of any version
+   *  - versions of iroha are organized in form of a tree: any new version has
+   *    only one previous version that itself can not be parent of the former
+   *    one.
+   *  - supported iroha versions are uniquely identified by 3 numbers: major,
+   *    minor and patch
+   *  - some different supported iroha versions may, and some require to be
+   *    compatible with some other versions
+   *  - over time, different irohad source and binaries can be tagged with same
+   *    supported version numbers, but the latter will be always children of the
```
Comment (ekovalev) [inline irohad/main/impl/pg_connection_init.cpp]: "latter" or "later"?
Also, "children" assumes direct "parent - child" relationship between two versions.
Maybe something like this:

> - if two different irohad sources and binaries are tagged with the same supported version number then the newer ones must be descendants of the older

File: irohad/main/irohad.cpp [hunk 2]
```diff
@@ -289,10 +292,16 @@ int main(int argc, char *argv[]) {
...
-            "overwritten. If you want overwrite ledger state, please "
-            "specify new genesis block using --genesis_block parameter.");
+        irohad.resetWsv();
+        if (not FLAGS_reuse_wsv) {
+          log->warn(
+              "No new genesis block is specified - blockstore will not be "
+              "overwritten. If you want overwrite ledger state, please "
+              "specify new genesis block using --genesis_block parameter. "
+              "If you want to reuse existing WSV data, consider the "
+              "--reuse_wsv "
```
[the remark this answers is not in the document]
Comment (MBoldyrev) [reply irohad/main/irohad.cpp]: wsv appears in [the docs](https://iroha.readthedocs.io/en/latest/concepts_architecture/architecture.html#world-state-view), but nevertheless I put it in braces after *state*

[the remark this answers is not in the document]
Comment (MBoldyrev) [reply irohad/main/irohad.cpp]: also changed the flag

File: irohad/main/irohad.cpp [hunk 3]
```diff
@@ -83,6 +83,8 @@ DEFINE_validator(keypair_name, &validate_keypair_name);
  */
 DEFINE_bool(overwrite_ledger, false, "Overwrite ledger data if existing");
 
+DEFINE_bool(reuse_state, false, "Try to use existing data in WSV at startup.");
```
Comment (ekovalev) [inline irohad/main/irohad.cpp]: Consider replacing the term "WSV" which may not be very clear to most users with a ubiquitous and self-explanatory "state"

**Comment by MBoldyrev** [comment]
migration script will be available in a separate repo shortly.

**Comment by MBoldyrev** [comment]
the migration script is here: https://github.com/hyperledger/iroha-state-migration-tool

**Pull request merged**